kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaProtocolHandler.java

```diff
@@ -526,25 +647,25 @@ public TransactionCoordinator initTransactionCoordinator(String tenant, PulsarAd
         TransactionConfig transactionConfig = TransactionConfig.builder()
                 .transactionLogNumPartitions(kafkaConfig.getTxnLogTopicNumPartitions())
                 .transactionMetadataTopicName(MetadataUtils.constructTxnLogTopicBaseName(tenant, kafkaConfig))
+                .brokerId(kafkaConfig.getBrokerId())
                 .build();
 
         MetadataUtils.createTxnMetadataIfMissing(tenant, pulsarAdmin, clusterData, kafkaConfig);
 
         TransactionCoordinator transactionCoordinator = TransactionCoordinator.of(
                 transactionConfig,
-                kafkaConfig.getBrokerId(),
+                brokerService.getPulsar().getClient(),
```

Comment:
Please rebase to master and use `SystemTopicClient`.

Reply:
Sure.

Reply:
Done. PTAL. Thanks.